contracts/two-party-pol-covenant/src/contract.rs

```diff
@@ -11,34 +13,51 @@ use covenant_astroport_liquid_pooler::msg::{
 use covenant_clock::msg::PresetClockFields;
 use covenant_ibc_forwarder::msg::PresetIbcForwarderFields;
 use covenant_interchain_router::msg::PresetInterchainRouterFields;
-use covenant_two_party_pol_holder::msg::{
-    PresetPolParty, PresetTwoPartyPolHolderFields, RagequitConfig,
-};
+use covenant_two_party_pol_holder::msg::{PresetTwoPartyPolHolderFields, RagequitConfig};
 use cw2::set_contract_version;
-use cw_utils::parse_reply_instantiate_data;
+use sha2::{Digest, Sha256};
 
 use crate::{
     error::ContractError,
-    msg::{InstantiateMsg, MigrateMsg, QueryMsg},
+    msg::{CovenantPartyConfig, InstantiateMsg, MigrateMsg, QueryMsg},
     state::{
         COVENANT_CLOCK_ADDR, COVENANT_POL_HOLDER_ADDR, LIQUID_POOLER_ADDR,
         PARTY_A_IBC_FORWARDER_ADDR, PARTY_A_ROUTER_ADDR, PARTY_B_IBC_FORWARDER_ADDR,
         PARTY_B_ROUTER_ADDR, PRESET_CLOCK_FIELDS, PRESET_HOLDER_FIELDS,
-        PRESET_LIQUID_POOLER_FIELDS, PRESET_PARTY_A_FORWARDER_FIELDS, PRESET_PARTY_A_ROUTER_FIELDS,
+        PRESET_PARTY_A_FORWARDER_FIELDS, PRESET_PARTY_A_ROUTER_FIELDS,
         PRESET_PARTY_B_FORWARDER_FIELDS, PRESET_PARTY_B_ROUTER_FIELDS,
     },
 };
 
 const CONTRACT_NAME: &str = "crates.io:covenant-two-party-pol";
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
-pub const CLOCK_REPLY_ID: u64 = 1u64;
-pub const HOLDER_REPLY_ID: u64 = 2u64;
-pub const PARTY_A_FORWARDER_REPLY_ID: u64 = 3u64;
-pub const PARTY_B_FORWARDER_REPLY_ID: u64 = 4u64;
-pub const LP_REPLY_ID: u64 = 5u64;
-pub const PARTY_A_ROUTER_REPLY_ID: u64 = 6u64;
-pub const PARTY_B_ROUTER_REPLY_ID: u64 = 7u64;
+pub const CLOCK_SALT: &[u8] = b"clock";
+pub const PARTY_A_INTERCHAIN_ROUTER_SALT: &[u8] = b"router_a";
+pub const PARTY_B_INTERCHAIN_ROUTER_SALT: &[u8] = b"router_b";
+pub const HOLDER_SALT: &[u8] = b"pol_holder";
+pub const PARTY_A_FORWARDER_SALT: &[u8] = b"forwarder_a";
+pub const PARTY_B_FORWARDER_SALT: &[u8] = b"forwarder_b";
+pub const LIQUID_POOLER_SALT: &[u8] = b"liquid_pooler";
+
+fn get_precomputed_address(
+    deps: Deps,
+    code_id: u64,
+    creator: &CanonicalAddr,
+    salt: &[u8],
+) -> Result<Addr, ContractError> {
+    let CodeInfoResponse { checksum, .. } = deps.querier.query_wasm_code_info(code_id)?;
+
+    let precomputed_address = instantiate2_address(&checksum, creator, salt)?;
+
+    Ok(deps.api.addr_humanize(&precomputed_address)?)
+}
+
+pub fn generate_contract_salt(salt_str: &[u8]) -> cosmwasm_std::Binary {
+    let mut hasher = Sha256::new();
+    hasher.update(salt_str);
+    hasher.finalize().to_vec().into()
+}
```

Comment:
Those helper function should be in "covenant-package", doesn't make sense to copy them around from contract to contract, will be hell to make changes to that in the future.